chore: Adds a users.uuid column

emrysal opened the pull request:
## What does this PR do?

null does not trigger unique constraint collisions, so the `@unique` is valid.
    
<!-- This is an auto-generated description by cubic. -->
---

## Summary by cubic
Adds a nullable uuid column to users with a unique index. New users get an auto-generated UUID; existing rows stay null until backfilled.

- **Migration**
  - If you need UUIDs for existing users, run a backfill after deploy: UPDATE "public"."users" SET uuid = gen_random_uuid() WHERE uuid IS NULL;

<!-- End of auto-generated description by cubic. -->

emrysal committed:
chore: Adds a users.uuid column

emrysal committed:
Add uuid backfill

emrysal committed:
Remove backfill due to possible locking issues

sean-brydon reviewed (changes requested):
We will need to backfill in batches

sean-brydon reviewed (dismissed):
LGTM

devin-ai-integration[bot] committed:
feat: Add batched UUID backfill to prevent database locking

Implements a batched approach for backfilling UUIDs on existing users:
- Processes 1000 rows at a time to avoid table locks
- Uses FOR UPDATE SKIP LOCKED to prevent blocking other transactions
- Includes small delays between batches (10ms) to allow other operations
- Logs progress every 10,000 rows for monitoring

This approach is safe for production databases with millions of users.

Co-Authored-By: alex@cal.com <me@alexvanandel.com>

sean-brydon commented on packages/prisma/migrations/20251023133244_add_user_uuid_column/migration.sql:
```diff
@@ -0,0 +1,11 @@
+
+*/
+-- AlterTable
+ALTER TABLE "public"."users" ADD COLUMN     "uuid" UUID;
```
This is indended.

sean-brydon committed:
Merge branch 'main' into chore/add-user-uuid-column
